modules/core/src/has_non_zero.simd.hpp

```diff
@@ -0,0 +1,279 @@
+    const v_type v_zero = vx_setzero_f32();
+    constexpr const int unrollCount = 8;
+    int step = v_type::nlanes * unrollCount;
+    int len0 = len & -step;
+    const float* srcSimdEnd = src+len0;
+
+    int countSIMD = static_cast<int>((srcSimdEnd-src)/step);
+    while(!res && countSIMD--)
+    {
+        v_type v0 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v1 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v2 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v3 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v4 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v5 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v6 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v7 = vx_load(src);
+        src += v_type::nlanes;
+        v0 |= v1;
+        v2 |= v3;
+        v4 |= v5;
+        v6 |= v7;
+
+        v0 |= v2;
+        v4 |= v6;
+        res = v_check_any(((v0 | v4) != v_zero));
+    }
+
+    v_cleanup();
+#endif
+    return res || hasNonZero_(src, srcEnd-src);
+}
+
+static bool hasNonZero64f( const double* src, size_t len )
+{
+    bool res = false;
+    const double* srcEnd = src+len;
+#if CV_SIMD_64F
+    typedef v_float64 v_type;
+    const v_type v_zero = vx_setzero_f64();
+    constexpr const int unrollCount = 16;
+    int step = v_type::nlanes * unrollCount;
+    int len0 = len & -step;
+    const double* srcSimdEnd = src+len0;
+
+    int countSIMD = static_cast<int>((srcSimdEnd-src)/step);
+    while(!res && countSIMD--)
+    {
+        v_type v0 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v1 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v2 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v3 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v4 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v5 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v6 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v7 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v8 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v9 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v10 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v11 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v12 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v13 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v14 = vx_load(src);
+        src += v_type::nlanes;
+        v_type v15 = vx_load(src);
+        src += v_type::nlanes;
+        v0  |= v1;
+        v2  |= v3;
+        v4  |= v5;
+        v6  |= v7;
+        v8  |= v9;
+        v10 |= v11;
+        v12 |= v13;
+        v14 |= v15;
+
+        v0  |= v2;
+        v4  |= v6;
+        v8  |= v10;
+        v12 |= v14;
+
+        v0  |= v4;
+        v8  |= v12;
+        res = v_check_any(((v0 | v8) != v_zero));
```

Comment:
-0 is not handled properly

Reply:
@chacha21 Friendly reminder.

Reply:
I just checked that `v_check_any(((v0 | v8) != v_zero));`  will be correct on a x64 target :

```#if CV_SIMD_64F
    typedef v_float64 v_type;
    const v_type v_zero = vx_setzero_f64();
    const v_type v_poszero = vx_setall_f64(0);
    const v_type v_negzero = vx_setall_f64(-0);
    const v_type v_one = vx_setall_f64(1);
    bool b1 = v_check_any((v_poszero != v_zero));//reports FALSE, as expected
    bool b2 = v_check_any((v_negzero != v_zero));//reports FALSE, as expected, because the _mm256_cmp_pd seems to handle -0 correctly
    bool b3 = v_check_any((v_one != v_zero));//reports TRUE, as expected
```

But I don't know if it behaves the same on NEON